media-gfx/krita/krita-5.2.2.ebuild

```diff
@@ -0,0 +1,134 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+ECM_TEST="forceoptional"
+PYTHON_COMPAT=( python3_{10..12} )
+KFMIN=5.82.0
+QTMIN=5.15.5
+VIRTUALX_REQUIRED="test"
+inherit ecm kde.org python-single-r1
+
+if [[ ${KDE_BUILD_TYPE} = release ]]; then
+	SRC_URI="mirror://kde/stable/${PN}/${PV}/${P}.tar.xz"
+	KEYWORDS="~amd64 ~arm64 ~ppc64 ~riscv ~x86"
+fi
+
+DESCRIPTION="Free digital painting application. Digital Painting, Creative Freedom!"
+HOMEPAGE="https://apps.kde.org/krita/ https://krita.org/en/"
+
+LICENSE="GPL-3"
+SLOT="5"
+IUSE="color-management fftw gif +gsl heif jpeg2k jpegxl +mypaint-brush-engine openexr pdf media +raw +xsimd webp"
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+
+# bug 630508
+RESTRICT="test"
+
+RDEPEND="${PYTHON_DEPS}
+	dev-libs/boost:=
+	dev-libs/quazip:0=[qt5(+)]
+	$(python_gen_cond_dep '
+		dev-python/PyQt5[declarative,gui,widgets,${PYTHON_USEDEP}]
+		dev-python/sip:=[${PYTHON_USEDEP}]
+	')
+	>=dev-qt/qtconcurrent-${QTMIN}:5
+	>=dev-qt/qtdbus-${QTMIN}:5
+	>=dev-qt/qtdeclarative-${QTMIN}:5
+	>=dev-qt/qtgui-${QTMIN}:5=[-gles2-only]
+	>=dev-qt/qtnetwork-${QTMIN}:5
+	>=dev-qt/qtprintsupport-${QTMIN}:5
+	>=dev-qt/qtsql-${QTMIN}:5
+	>=dev-qt/qtsvg-${QTMIN}:5
+	>=dev-qt/qtwidgets-${QTMIN}:5
+	>=dev-qt/qtx11extras-${QTMIN}:5
+	>=dev-qt/qtxml-${QTMIN}:5
+	>=kde-frameworks/kcompletion-${KFMIN}:5
+	>=kde-frameworks/kconfig-${KFMIN}:5
+	>=kde-frameworks/kcoreaddons-${KFMIN}:5
+	>=kde-frameworks/kcrash-${KFMIN}:5
+	>=kde-frameworks/kguiaddons-${KFMIN}:5
+	>=kde-frameworks/ki18n-${KFMIN}:5
+	>=kde-frameworks/kiconthemes-${KFMIN}:5
+	>=kde-frameworks/kitemmodels-${KFMIN}:5
+	>=kde-frameworks/kitemviews-${KFMIN}:5
+	>=kde-frameworks/kwidgetsaddons-${KFMIN}:5
+	>=kde-frameworks/kwindowsystem-${KFMIN}:5
+	>=kde-frameworks/kxmlgui-${KFMIN}:5
+	media-gfx/exiv2:=
+	media-libs/lcms
+	media-libs/libjpeg-turbo:=
+	media-libs/libpng:=
+	media-libs/tiff:=
+	sys-libs/zlib
+	virtual/opengl
+	x11-libs/libX11
+	x11-libs/libXi
+	color-management? ( >=media-libs/opencolorio-2.0.0 )
+	fftw? ( sci-libs/fftw:3.0= )
+	gif? ( media-libs/giflib )
+	gsl? ( sci-libs/gsl:= )
+	jpeg2k? ( media-libs/openjpeg:= )
+	jpegxl? ( >=media-libs/libjxl-0.7.0_pre20220825 )
+	heif? ( >=media-libs/libheif-1.11:=[x265] )
+	mypaint-brush-engine? ( media-libs/libmypaint:= )
+	openexr? ( media-libs/openexr:= )
+	pdf? ( app-text/poppler[qt5] )
+	media? ( media-libs/mlt:= )
+	raw? ( kde-apps/libkdcraw:= )
+	xsimd? ( dev-cpp/xsimd )
+	webp? ( >=media-libs/libwebp-1.2.0:= )
+
+"
+DEPEND="${RDEPEND}"
+BDEPEND="
+	dev-cpp/eigen:3
+	dev-lang/perl
+	sys-devel/gettext
+	dev-libs/immer
+	dev-libs/lager
+	dev-libs/zug
+	dev-libs/libunibreak
+"
+
+PATCHES=(
+	"${FILESDIR}/${PN}-4.3.1-tests-optional.patch"
+	"${FILESDIR}/${PN}-5.1.5-sip-6.8.patch" # bug 919139
+	"${FILESDIR}/${PN}-5.1.5-libjxl-0.9.patch" # bug 922524
+	"${FILESDIR}/${PN}-5.2.2-fftw.patch"
+	"${FILESDIR}/${PN}-5.2.2-openjpeg.patch"
```

Comment:
What are those about? They are new compared to 5.1.5 and the files make no mention of why disabling finding these packages by patch is necessary.

Reply:
Basically the build system tries to look for cmake provided by those packages first before falling back to pkgconfig. I've added comments about this in the patch files to make this clear.